src/main/java/io/nats/client/impl/Headers.java

```diff
@@ -390,6 +394,40 @@ public ByteArrayBuilder appendSerialized(ByteArrayBuilder bab) {
 		return bab;
 	}
 
+	/**
+	 * Write the header to the byte array. Assumes that the caller has
+	 * already validated that the destination array is large enough by using getSerialized()
+	 * @param destPosition the position index in destination byte array to start
+	 * @param dest the byte array to write to
+	 * @return the length of the header
+	 */
+	public int serializeToArray(int destPosition, byte[] dest) {
+		System.arraycopy(HEADER_VERSION_BYTES_PLUS_CRLF, 0, dest, destPosition, HVCRLF_BYTES);
+		destPosition += HVCRLF_BYTES;
+
+		for (Map.Entry<String, List<String>> entry : valuesMap.entrySet()) {
+			List<String> values = entry.getValue();
+			for (String value : values) {
+				byte[] bytes = entry.getKey().getBytes(US_ASCII);
+				System.arraycopy(bytes, 0, dest, destPosition, bytes.length);
+				destPosition += bytes.length;
+
+				dest[destPosition++] = COLON;
+
+				bytes = value.getBytes(US_ASCII);
+				System.arraycopy(bytes, 0, dest, destPosition, bytes.length);
+				destPosition += bytes.length;
+
+				dest[destPosition++] = CR;
+				dest[destPosition++] = LF;
+			}
+		}
+		dest[destPosition++] = CR;
+		dest[destPosition] = LF;
```

Comment:
is the `destPosition++` missing here on purpose?

Reply:
Yeah. It's the last usage, IntelliJ reminded me that destPosition isn't used any more. I'm returning the length of the data, not the position.